src/main/webapp/app/course/learning-paths/participate/learning-path-storage.service.ts

```diff
@@ -0,0 +1,211 @@
+    }
+
+    /**
+     * Returns if the learning path's history stores at least one entry.
+     *
+     * @param learningPathId the id of the learning path for which the history should be checked
+     */
+    hasPrevious(learningPathId: number): boolean {
+        if (this.learningPathHistories.has(learningPathId)) {
+            return this.learningPathHistories.get(learningPathId)!.length !== 0;
+        }
+        return false;
+    }
+
+    /**
+     * Gets and removes the latest stored entry from the learning path's history.
+     *
+     * @param learningPathId
+     */
+    getPrevious(learningPathId: number) {
+        if (!this.hasPrevious(learningPathId)) {
+            return undefined;
+        }
+        return this.learningPathHistories.get(learningPathId)!.pop();
+    }
+
+    /**
+     * Simplifies and stores the recommended order of learning objects for the given learning path
+     *
+     * @param learningPathId the id of the learning path
+     * @param learningPath the learning path dto that should be stored
+     */
+    storeRecommendations(learningPathId: number, learningPath: NgxLearningPathDTO) {
+        this.learningPathRecommendations.set(learningPathId, []);
+        let currentId = learningPath.nodes.map((node) => node.id).find((id) => !learningPath.edges.find((edge) => edge.target == id));
+        while (currentId) {
+            const currentNode = learningPath.nodes.find((node) => node.id == currentId)!;
+            if (currentNode.type === NodeType.LECTURE_UNIT) {
+                this.learningPathRecommendations.get(learningPathId)!.push(new LectureUnitEntry(currentNode.linkedResourceParent!, currentNode.linkedResource!));
+            } else if (currentNode.type === NodeType.EXERCISE) {
+                this.learningPathRecommendations.get(learningPathId)!.push(new ExerciseEntry(currentNode.linkedResource!));
+            }
+            const edge = learningPath.edges.find((edge) => edge.source == currentId);
+            if (edge) {
+                currentId = edge.target;
+            } else {
+                currentId = undefined;
+            }
+        }
+    }
+
+    /**
+     * Returns if there is a recommendation left that has not been interacted with yet.
+     *
+     * @param learningPathId the id of the learning path
+     */
+    hasRecommendation(learningPathId: number) {
+        if (this.learningPathRecommendations.has(learningPathId)) {
+            return this.learningPathRecommendations.get(learningPathId)!.find((entry) => !entry.interacted) !== undefined;
+        }
+        return false;
+    }
+
+    /**
+     * Sets the given entry interaction.
+     *
+     * @param learningPathId the id of the learning path the entry belongs to
+     * @param entry the entry that should be set to interacted with
+     * @param value the value that should be set
+     */
+    setInteraction(learningPathId: number, entry: StorageEntry, value: boolean) {
```

Comment:
Can you give `value` a better name? Is that supposed to mean `visited` or something like that?